src/astarte-charts/react/hooks/useChartProviders.ts

```diff
@@ -0,0 +1,89 @@
+/*
+   This file is part of Astarte.
+
+   Copyright 2020 Ispirata Srl
+
+   Licensed under the Apache License, Version 2.0 (the "License");
+   you may not use this file except in compliance with the License.
+   You may obtain a copy of the License at
+
+      http://www.apache.org/licenses/LICENSE-2.0
+
+   Unless required by applicable law or agreed to in writing, software
+   distributed under the License is distributed on an "AS IS" BASIS,
+   WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+   See the License for the specific language governing permissions and
+   limitations under the License.
+*/
+
+import { useCallback, useEffect, useState } from 'react';
+import type { ChartProvider, ChartData, ChartDataWrapper, ChartDataKind } from 'astarte-charts';
+
+type Status = 'initial' | 'loading' | 'loaded';
+
+type DataFetcher<Data> =
+  | {
+      status: 'initial';
+      data: null;
```

Comment:
you have "initial" as constant to discriminate this state. `data: null` is unnecessary

Reply:
That's a good point and I'd like your opinion on it. Conceptually, I agree. But I am wondering if leaving the property as `null` won't be that harmful and instead allow for more agile code. I'll explain.
- With `data: null`, a React component can always access the property and directly discriminates whether data is present or not.
- If instead the component is forced to discriminate over `status` before accessing the `data` property, that would mean re-rendering the component when entering both `loading` and `loaded` statuses. That's unneeded work if the component only cares about `data != null`.